docs/DeveloperGuide.adoc

```diff
@@ -237,6 +238,57 @@ image::CommitActivityDiagram.png[]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
+// tag::folders[]
+=== Folders feature
+==== Implementation
+
+A bookmark can be in a folder, and a folder can be nested within other folders for traditional directory organization.
+
+This mechanism is facilitated mainly by `Folder` and `FolderStructure`.
+
+`Folder` is simply another field on `Bookmark`, just like `Url` or `Name`, and has a single `String` property that contains the folder the bookmark is located.
+
+`FolderStructure` represents the hierarchy of folders, containing the folder it represents and its subfolders
+
+.Class Diagram of the Folders component
+image::FolderStructureClassDiagram.png[]
+
+
+Given below is an example usage scenario and how the folder structure behaves at each step.
+
+Step 1. The user launches the application for the first time. The root `FolderStructure` will be initialized with the initial hierarchy in the stored data.
+
+image::FolderStructureState0.png[]
+
+Step 2. The user executes `folder GER1000 p/Work` command to create a new `FolderStructure` in the subfolders of `work`. Starting from the `root`, a depth first search will be performed to locate `work`. When found, `ger1000` will be added to its subfolders.
+
+image::FolderStructureState1.png[]
+
+[NOTE]
+If the parent folder is not provided, the parent folder will default to `root`.
+
+The following sequence diagram shows how the folder operation works:
+
+image::FolderSequenceDiagram.png[]
+
+NOTE: The lifeline for `AddFolderCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+Finally, the user sees his folder added successfully in the folder hierarchy.
+
+The following activity diagram summarizes what happens when a user executes a new command:
```

Comment:
Is this missing diagram a case of not-done-yet, or forgot-to-include? Haha if it's the former, maybe can add a comment about what it's meant to be?

Reply:
Nah there's no activity diagram, it's kind of a linear thing unlike the overarching undo/redo